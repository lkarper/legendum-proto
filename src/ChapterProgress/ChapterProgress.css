.ChapterProgress__link-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    border: 2px solid #FFC71F;
    background-color: #721D2B;
    padding: 5px;
    width: 90%;
    margin: 5px auto;
}

.ChapterProgress__link {
    color: #FFC71F;
    text-decoration: none;
    font-family: 'EB Garamond', serif;
    margin: 5px 10px;
    font-size: 20px;
    text-align: center;
}

.ChapterProgress__link:active {
    text-decoration: underline;
}

.ChapterProgress__link:hover {
    color: goldenrod;
}

.ChapterProgress__leaf {
    margin: 5px;
    color: #FFC71F;
}

.ChapterProgress__h4 {
    font-size: 22px;
}

.ChapterProgress__h5 {
    font-size: 20px;
}

@media screen and (min-width: 1000px) {
    .ChapterProgress__link-container {
        width: 75%;
        margin-top: 15px;
    }

    .ChapterProgress__link {
        font-size: 20px;
    }

    .ChapterProgress__li-dates-completed,
    .ChapterProgress__p-times-completed {
        font-size: 20px;
    }

    .ChapterProgress__h4 {
        font-size: 24px;
    }

    .ChapterProgress__h5 {
        font-size: 22px;
    }
}
