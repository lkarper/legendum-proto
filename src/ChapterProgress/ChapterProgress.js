import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import UserContext from '../contexts/UserContext';

const ChapterProgress = (props) => {

    const context = useContext(UserContext);
    const { exercise } = props;

    const timesCompleted = context.progress.filter(p => p.exercise_id === exercise.id).length;

    return (
        <li>
            <h3>{exercise.exercise_title}{' '}{exercise.exercise_translation}</h3>
            <p>You've completed this chapter {timesCompleted === 1 ? '1 time' : `${timesCompleted} times`}</p>
            {timesCompleted ? 
                <div> 
                    <h4>You completed this exercise on: </h4>
                    <ul>
                        {context.progress
                            .filter(p => p.exercise_id === exercise.id)
                            .map(p => <li key={p.id}>{new Date(p.date_completed).toString()}</li>)
                        }
                    </ul>
                </div>
                : ''
            }
            <Link to={`/game/exercises/${exercise.id}/learn`}>Learn this Exercise</Link>
            <Link to={`/game/exercises/${exercise.id}/do`}>Practice this Exercise</Link>
        </li>
    );
}

export default ChapterProgress;