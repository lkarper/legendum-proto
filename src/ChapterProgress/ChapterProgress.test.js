import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ChapterProgress from './ChapterProgress';

describe('ChapterProgress component', () => {
  it('renders the UI without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <ChapterProgress />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const wrapper = mount(
      <BrowserRouter>
        <ChapterProgress />
      </BrowserRouter>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
