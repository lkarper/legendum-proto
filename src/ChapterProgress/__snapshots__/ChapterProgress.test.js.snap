// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChapterProgress component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ChapterProgress
      exercise={Object {}}
    >
      <li>
        <h4>
          Error
        </h4>
        <p>
          Looks like something went wrong. Check your connection and try again.
        </p>
      </li>
    </ChapterProgress>
  </Router>
</BrowserRouter>
`;
